Library/Homebrew/compat/download_strategy.rb

```diff
@@ -0,0 +1,26 @@
+# typed: false
+# frozen_string_literal: true
+
+class AbstractDownloadStrategy
+  module Compat
+    def fetch(timeout: nil)
+      super()
+    end
+  end
+
+  class << self
+    def method_added(method)
+      if method == :fetch && instance_method(method).arity.zero?
+        odeprecated "`def fetch` in a subclass of #{self}",
+                    "`def fetch(timeout: nil, **options)` and output a warning " \
+                    "when `options` contains new unhandled options"
```

Comment:
We had more complaints than this, such as `_fetch` and `resolve_url_basename_time_file_size`.

Reply:
These are definitely private APIs. The docs specifically mention `def fetch`, nothing else.

Reply:
Regardless, I think it would be good to handle the `_fetch` case at least.

Reply:
Added a deprecation when calling `_fetch` without `timeout`.